### DISCUSSION TITLE
How to build Model Variants

### DISCUSSION THREAD

**Discussion by adrianposteuca** [google/model-viewer#3273] [Q&A] [answered]
Hello guys,

I saw that is possible to use model variants. I am trying to build this option but I do not know where to start.
I have the 3d model with some textures but how can I embed new variants, and then export the new model?

Thank you
Adrian

**Comment by samaneh-kazemi** [comment]
Hi @adrianposteuca, have you looked at our [variants example](https://modelviewer.dev/examples/scenegraph/#variants)? I recommend playing around with that in glitch.

**Comment by elalish** [comment, marked as answer]
Our [editor](https://modelviewer.dev/editor/) has this functionality. Go the materials tab and create a variant and upload new textures for it, then download the scene.

**Reply by plutorkhan** [reply]
Hi, I'm trying to add a couple of materials to the model but the editor doesn't export the model after adding variants to it. The console logs:

Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'material')
    at space-opera.js:88772:60
    at Mesh.traverse (space-opera.js:33845:3)
    at Group.traverse (space-opera.js:33851:18)
    at GLTFExporterMaterialsVariantsExtension.beforeParse (space-opera.js:88763:20)
    at space-opera.js:88132:27
    at GLTFWriter._invokeAll (space-opera.js:88178:4)
    at GLTFWriter.processInput (space-opera.js:88130:8)
    at GLTFWriter.write (space-opera.js:86499:8)
    at GLTFExporter.parse (space-opera.js:86187:10)
    at space-opera.js:90230:26

**Reply by elalish** [reply]
Can you file an issue with specific repro instructions? This sounds like a bug.

**Reply by plutorkhan** [reply]
done
